src/pages/Login/index.jsx

```diff
@@ -93,12 +91,25 @@ const LoginScreen = () => {
           const usernameDoc = db.collection(`users`);
           await usernameDoc.doc(auth.currentUser.uid).set({
             uid: val.user.uid,
+            username: val.user.uid,
             name: val.user.displayName,
             photoURL: val.user.photoURL,
             displayName: val.user.displayName,
             Friends: [],
             posts: [],
           });
+        } else {
```

Comment:
I was thinking, if we make sure to add usernames in signup only we won't need to query for username and give it a default everytime if it's a old sign up as this is a costly operation, what do you say?

Reply:
But, what if the user is new and they tried to sign up through login page. I mean they are new and they click on the google sign in.

Reply:
I meant, in the else condition where existing users sign in, we're querying if username is set or not... and that won't be needed as they already had signed up with username

Reply:
alright, I'm letting this fix persist and will remove it after clearing users without usernames